src/server/game/Server/Packets/MailPackets.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2008-2015 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "MailPackets.h"
+
+void WorldPackets::Mail::MailGetList::Read()
+{
+    _worldPacket >> Mailbox;
+    LowGuids.resize(_worldPacket.read<int32>());
+
+    for (auto l : LowGuids)
+        _worldPacket >> l;
+}
+
+WorldPacket const* WorldPackets::Mail::MailListResult::Write()
+{
+    _worldPacket << int32(Mails.size());
+    _worldPacket << int32(TotalNumRecords);
+
+    for (auto const& mail : Mails)
+    {
+        _worldPacket << int32(mail.MailID);
+        _worldPacket << uint8(mail.SenderType);
+
+        _worldPacket.WriteBit(mail.SenderHint.VirtualRealmAddress.HasValue);
+        _worldPacket.WriteBit(mail.SenderHint.NativeRealmAddress.HasValue);
+        _worldPacket.FlushBits();
+
+        if (mail.SenderHint.VirtualRealmAddress.HasValue)
+            _worldPacket << int32(mail.SenderHint.VirtualRealmAddress.Value);
+
+        if (mail.SenderHint.NativeRealmAddress.HasValue)
+            _worldPacket << int32(mail.SenderHint.NativeRealmAddress.Value);
+
+        _worldPacket << int64(mail.Cod);
+        _worldPacket << int32(mail.PackageID);
+        _worldPacket << int32(mail.StationeryID);
+        _worldPacket << int64(mail.SentMoney);
+        _worldPacket << int32(mail.Flags);
+        _worldPacket << float(mail.DaysLeft);
+        _worldPacket << int32(mail.MailTemplateID);
+
+        _worldPacket << int32(mail.Attachments.size());
+
+        for (auto const& att : mail.Attachments)
+        {
+            _worldPacket << uint8(att.Position);
+            _worldPacket << int32(att.AttachID);
```

Comment:
pretty sure this one is uint64 in live builds (magic build didnt use 128 bit guids)